docs/design/ha_master.md

```diff
@@ -0,0 +1,188 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<!-- TAG RELEASE_LINK, added by the munger automatically -->
+<strong>
+The latest release of this document can be found
+[here](http://releases.k8s.io/release-1.3/docs/design/access.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Automated HA master deployment (in GCE)
+
+**Author:** filipg@, jsz@
+
+# Introduction
+We want to allow users to easily replicate kubernetes masters to have highly available cluster,
+using automated tools like kube-up.sh and kube-down.sh.
+
+This document describes technical design of this feature. It assumes that we are using aforementioned
+scripts for cluster deployment. It focuses on GCE use-case but all of the ideas described in
+the following sections should be easy to port to AWS, other cloud providers and bare-metal environment.
+
+# Overview
+In a cluster with replicated master machine we will have the following component interactions:
+* All etcd replicas will be clustered together and will be using master election
+and quorum mechanism to agree on the state. All of these mechanisms are integral
+parts of etcd and we will only have to configure them properly.
+* All apiserver replicas will be working independently talking to a etcd on the
+same machine, which if needed will forward requests to the current etcd master
```

Comment:
It would be nice to have an assumptions and clauses section at the top.  I think that would help the reader.